File: Progetto-2023-2024-Vecchio-Larocca-Lentini-Mahrous/backend/src/main/java/it/unical/inf/ea/backend/data/services/interfaces/AddressService.java
package it.unical.inf.ea.backend.data.services.interfaces;

import it.unical.inf.ea.backend.dto.AddressDTO;
import it.unical.inf.ea.backend.dto.creation.AddressCreateDTO;

import java.util.List;
import java.util.UUID;

public interface AddressService {

    AddressDTO createAddress(AddressCreateDTO addressCreateDTO) throws IllegalAccessException;
    AddressDTO updateAddress(UUID id, AddressDTO addressDTO) throws IllegalAccessException;
    void setDefaultAddress(UUID id) throws IllegalAccessException;
    void deleteAddress(UUID id) throws IllegalAccessException;
    AddressDTO getAddressById(UUID id) throws IllegalAccessException;
    List<AddressDTO> getAllAddresses();
    List<AddressDTO> getAllLoggedUserAddresses();
}

File: Progetto-2023-2024-Vecchio-Larocca-Lentini-Mahrous/frontend/app/src/main/java/com/android/frontend/service/AddressService.kt
// ShippingService.kt
package com.android.frontend.service

import com.android.frontend.dto.creation.AddressCreateDTO
import com.android.frontend.dto.AddressDTO
import retrofit2.Call
import retrofit2.http.*

interface AddressService {

    @POST("/api/v1/addresses/addAddress")
    fun addAddress(
        @Header("Authorization") authorization: String,
        @Body addressCreateDTO: AddressCreateDTO,
    ): Call<AddressDTO>

    @DELETE("/api/v1/addresses/deleteAddress/{id}")
    fun deleteAddress(
        @Header("Authorization") authorization: String,
        @Path("id") id: String
    ): Call<Void>

    @PUT("/api/v1/addresses/setDefaultAddress/{id}")
    fun setAddress(
        @Header("Authorization") authorization: String,
        @Path("id") id: String
    ): Call<Void>

    @PUT("/api/v1/addresses/updateAddress/{id}")
    fun updateAddress(
        @Header("Authorization") authorization: String,
        @Path("id") id: String,
        @Body shippingAddressDTO: AddressDTO
    ): Call<AddressDTO>

    @GET("/api/v1/addresses/getAddress/{id}")
    fun getShippingAddress(
        @Header("Authorization") authorization: String,
        @Path("id") id: String
    ): Call<AddressDTO>

    @GET("/api/v1/addresses/getAllAddresses")
    fun getAllShippingAddresses(
        @Header("Authorization") authorization: String
    ): Call<List<AddressDTO>>

    @GET("/api/v1/addresses/getAllLoggedUserAddresses")
    fun getAllLoggedUserAddresses(
        @Header("Authorization") authorization: String
    ): Call<List<AddressDTO>>
}
